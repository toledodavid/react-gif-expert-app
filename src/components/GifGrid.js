import React, {useEffect} from 'react';


const GifGrid = ({category}) => {

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Rick&limit=10&api_key=FMH46HzHk8wFS4OcWPBUXvpm7eYt14bF';

    const response = await fetch(url);
    const {data} = await response.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        imgUrl: img.images?.downsized_medium.url
      }
    });
    console.log(gifs);
  }


  return(
    <div>
      <h3>{category}</h3>
    </div>
  );
}


export default GifGrid;