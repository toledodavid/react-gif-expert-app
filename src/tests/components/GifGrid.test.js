import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Tests for <GifGrid /> component', () => {

  const category = 'Superman';

  test('It should render correctly', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show items when images have been loaded from useFetchGifs', () => {
    const gifs = [
      {
        id: '121',
        title: 'TestTitle',
        imgUrl: 'http://test.com/example'
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').exists()).toBe(true);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });

});