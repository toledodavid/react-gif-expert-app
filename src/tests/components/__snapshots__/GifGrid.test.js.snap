// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGrid /> component It should render correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Superman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests for <GifGrid /> component It should show items when images have been loaded from useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Superman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="121"
      imgUrl="http://test.com/example"
      key="121"
      title="TestTitle"
    />
  </div>
</Fragment>
`;
